contrib/devtools/previous-release.py

```diff
@@ -0,0 +1,207 @@
+import sys
+import hashlib
+
+def usage(args, extras):
+    print('Usage: {} [options] tag1 [tag2..tagN]'.format(sys.argv[0]))
+    print('Specify release tag(s), e.g.: {} v0.18.1'.format(sys.argv[0]))
+    return 0
+
+@contextlib.contextmanager
+def pushd(new_dir):
+    previous_dir = os.getcwd()
+    os.chdir(new_dir)
+    try:
+        yield
+    finally:
+        os.chdir(previous_dir)
+
+def download_binary(tag, args):
+    if Path(tag).is_dir():
+        if not args.remove_dir:
+            print('Using cached {}'.format(tag))
+            return 0
+        shutil.rmtree(tag)
+    Path(tag).mkdir()
+    bin_path = 'bin/bitcoin-core-{}'.format(tag[1:])
+    match = re.compile('v(.*)(rc[0-9]+)$').search(tag)
+    if match:
+        bin_path = 'bin/bitcoin-core-{}/test.{}'.format(match.group(1),
+                                                        match.group(2))
+    tarball = 'bitcoin-{tag}-{platform}.tar.gz'.format(tag=tag[1:],
+                                                       platform=args.platform)
+    sha256Sums = "SHA256SUMS-{tag}.asc".format(tag=tag[1:])
+    tarballUrl = 'https://bitcoincore.org/{bin_path}/{tarball}'.format(bin_path=bin_path,
+                                                            tarball=tarball)
+    sha256SumsUrl = 'https://bitcoincore.org/{bin_path}/SHA256SUMS.asc'.format(bin_path=bin_path)
+
+    print('Fetching: {tarballUrl}'.format(tarballUrl=tarballUrl))
+    print('Fetching: {sha256SumsUrl}'.format(sha256SumsUrl=sha256SumsUrl))
+
+    header, status = subprocess.Popen(['curl', '-I', tarballUrl], stdout=subprocess.PIPE).communicate()
+    if re.search("404 Not Found",header.decode("utf-8")):
+        print("tag for binary does not exist")
+        return 1
+
+    curlCmds = [
+        ['curl', '-O', tarballUrl],
+        ['curl', "-o", sha256Sums, sha256SumsUrl],
+    ]
+
+    for cmd in curlCmds:
+        ret = subprocess.call(cmd)
+        if ret:
+            return ret
+
+    hasher = hashlib.sha256()
+    with open(tarball, "rb", encoding="utf-8") as afile:
+        buf = afile.read()
+        hasher.update(buf)
+    afile.close()
+    tarballHash = hasher.hexdigest()
+    file = open(sha256Sums, 'r', encoding="utf-8")
+    lst = list(file.readlines())
+    file.close()
+    lastline = lst[len(lst)-1]
+
+    for line in lst:
+        if re.search(tarballHash,line):
+            break
+        elif lastline == line:
+            print("Hash does not match")
+            Path(tarball).unlink()
+            return 1
+
+    ret = subprocess.call([ 'tar', '-zxf', tarball, '-C', tag, '--strip-components=1',
+            'bitcoin-{tag}'.format(tag=tag[1:]) ])
+    if ret:
+        return ret
+
+    Path(tarball).unlink()
+    Path(sha256Sums).unlink()
+    return 0
+
+def build_release(tag, args):
+    if args.remove_dir:
+        if Path(tag).is_dir():
+            shutil.rmtree(tag)
+    if not Path(tag).is_dir():
+        output = subprocess.check_output(['git', 'tag', '-l', tag])
```

Comment:
Could use `subprocess.run(..., check=True)` here and where you're using `subprocess.call()` etc; as opposed to the older subprocess API.

Reply:
I changed the other `subprocess.call` methods to `subprocess.run` but when I used `run` for this line it hung up the script. I had to exit out of the tag list to continue the script. However `check_output` worked.